### ISSUE TITLE
Minification failed with error

### ISSUE THREAD

**Issue by pg-ey** [SAP/ui5-tooling#890]
## Expected Behavior
Build successful with minification

## Current Behavior
Error Message:
Minification failed with error: 'return' outside of function in file get-own-property-symbols.js (line 11, col 1, pos 255)

## Steps to Reproduce the Issue

1.- name: ui5-task-minifier
      afterTask: minify
      configuration:
        html: true
        css: true
        json: true

## Context
 * UI5 Module Version (output of `ui5 --version` when using the CLI): 3.6.0
 * Node.js Version: 18.14.2
 * npm Version: 9.5.2
 * OS/Platform: Windows 10
 * Browser *(if relevant)*: Chrome

## Log Output / Stack Trace
Error: Minification failed with error: 'return' outside of function in file get-own-property-symbols.js (line 11, col 1, pos 255)
    at AsyncFunction.execMinification (file:///home/user/projects/bre-v3/app/node_modules/@ui5/cli/node_modules/@ui5/builder/lib/processors/minifierWorker.js:63:9)

**Comment by flovogt** [comment]
Hi @pg-ey,
thanks for opening this issue here. From the input you provide above I expect a `return` statement in `get-own-property-symbols.js` which is not correctly placed. This file might be a custom js file located in your application and not provided by an UI5 library.
Could you please check this file?
If the issue still appears after you have fixed the `return` statement. Feel free to share the project with us, so we can have a deeper look.

**Comment by pg-ey** [comment]
Hi @flovogt , thanks for your response.
I don't have any custom js file with the name get-own-property-symbols.js.
However, when I build my app excluding minify, it works fine.

I have a doubt that there is a reference to getOwnPropertySymbols in one of the @ui5/cli dependencies which is causing the issue at the time of minifying.

**Comment by flovogt** [comment]
Could you share your `ui5.yaml` and `package.json` or even better the whole application?

**Comment by flovogt** [comment]
Do you have included https://github.com/es-shims/get-own-property-symbols into your project?

**Comment by pg-ey** [comment]
> Do you have included https://github.com/es-shims/get-own-property-symbols into your project?

No, I haven't included this.

**Comment by pg-ey** [comment]
> Could you share your `ui5.yaml` and `package.json` or even better the whole application?

Adding package.json and ui5.yaml here
[package (1).json](https://github.com/SAP/ui5-tooling/files/13175905/package.1.json)
[ui5.zip](https://github.com/SAP/ui5-tooling/files/13175914/ui5.zip)

**Comment by flovogt** [comment]
1. Remove the `ui5` section in the package.json - not needed any longer with UI5 Tooling v3
2. I wonder if you need `@sap/ui5-builder-webide-extension` at all. Could please remove this dependency and also the tasks used in the ui5.yaml and see if the build works as expected

**Comment by RandomByte** [comment]
Also, could you please share the full build log? Ideally after adding the `--verbose` parameter.

**Comment by pg-ey** [comment]
> 1. Remove the `ui5` section in the package.json - not needed any longer with UI5 Tooling v3
> 2. I wonder if you need `@sap/ui5-builder-webide-extension` at all. Could please remove this dependency and also the tasks used in the ui5.yaml and see if the build works as expected

Removing the 'ui5' section in the package.json has worked.
I will keep the  `@sap/ui5-builder-webide-extension` till I figure out an alternative for my custom tasks.

Thanks @flovogt for the help.

**Comment by VinayCh0wdary** [comment]
I'm having the same issue, I did try removing the ui5 section in the package.json getting the same error even doing that.
![Screenshot 2024-08-06 at 6 02 13 PM](https://github.com/user-attachments/assets/4a92aeac-c5c4-4aa3-a4bb-57b8ec9b5aa3)

**Comment by flovogt** [comment]
> I'm having the same issue, I did try removing the ui5 section in the package.json getting the same error even doing that. ![Screenshot 2024-08-06 at 6 02 13 PM](https://private-user-images.githubusercontent.com/94469489/355448757-4a92aeac-c5c4-4aa3-a4bb-57b8ec9b5aa3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NTM3OTMsIm5iZiI6MTcyMjk1MzQ5MywicGF0aCI6Ii85NDQ2OTQ4OS8zNTU0NDg3NTctNGE5MmFlYWMtYzVjNC00YWEzLWE0YmItNTdiOGVjOWI1YWEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDE0MTEzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmMDhlMmNhZDE2OTZjNmRjYjg0MzMwMGNhODJiYzZiZmE5ZGY0OWFmMjVhNWU2OWJkMjIxODkxYjhlYWRiMmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._obH87rNWDe9zj7RekcLd636yc4xBccT9gEtpC1YIbg)


@VinayCh0wdary Please open a new issue and share your project details with us, so we can have a deeper analysis.

**Comment by Vest** [comment]
@VinayCh0wdary I think, I know where the error comes from. It seems that the builder is stupid, but straightforward - it scans all the *.js files in webapp/. If (by any chance) you have a node.js application there with node_modules/ inside, the minifier tries to process all *.js files. And if they don't meet minifier's rules (e.g., there must be no [return;](https://github.com/inspect-js/has-symbols/blob/270f0a67b5f4cb12715e0a84e672aaee64d1b4f1/test/shams/get-own-property-symbols.js#L12) in the global scope), it crashes with a useless error.

The error handling will be better if it display me the file path and not just a file name. After I debugged it, I found out that you have to exclude node.js resources from the builder folder. I have fixed my error (there was a package has-symbols in /webapp/.../node_modules/...) with the following configuration in ui5.yaml:
```yaml
builder:
    resources:
        excludes:
            - "**/node_modules/**"
            - "**/localService/**"
```

**Comment by flovogt** [comment]
Thanks a lot @Vest for your input. I can confirm there should be no "node_modules" inside the "webapp" folder. In most scenarios I am aware of having a "node_modules" folder inside the "webapp" folder happens unintended. So, it should be removed.

**Comment by RandomByte** [comment]
> The error handling will be better if it display me the file path and not just a file name.

@Vest thanks for the suggestion, we'll fix this in one of the next releases via https://github.com/SAP/ui5-builder/pull/1102 👍

**Comment by Vest** [comment]
@flovogt I didn't know this, before I executed `ui5 build`. I have created a nodejs module in webapp/ because I developed a small proxy as a part of localService/ (it contained sample data for JS-based MockServer). That's why I decided to simply have my pet-proxy there.
I know that it doesn't sound right to have node_modules inside... otherwise, I wouldn't help you with this bug :smile:

@RandomByte thanks for the feature. The more verbose error is, the easier will it be to solve it.
